### PR TITLE
move to use dir.build.props

### PR DESCRIPTION
**Pull request by jennyf19**
Will make it easier to add the new dlls for the dev ex.

### CODE REVIEW

File: src/Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    <RepositoryUrl>https://github.com/AzureAD/microsoft-identity-web</RepositoryUrl>
+    <PackageReleaseNotes>The release notes are available at https://github.com/AzureAD/microsoft-identity-web/releases and the roadmap at https://github.com/AzureAD/microsoft-identity-web/wiki#roadmap </PackageReleaseNotes>
+    <PackageTags>Microsoft Identity Web UI;Microsoft identity platform;Microsoft Identity Web;.NET;ASP.NET Core;Web App;Web API;B2C;Azure Active Directory;AAD;Identity;Authentication;Authorization</PackageTags>
+    <DefineConstants>$(DefineConstants);WEB</DefineConstants>
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <TargetFrameworks>netcoreapp3.1; net462; net472; net5.0; netstandard2.0</TargetFrameworks>
+    <SignAssembly>true</SignAssembly>
```
Comment (jmprieur) [inline src/Directory.Build.props:25]: Duplicate of line 9

File: src/Directory.Build.props [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    <AdditionalFiles Include="..\..\stylecop.json" />
+  </ItemGroup>
+
+  <ItemGroup Label="Build Tools" Condition="$([MSBuild]::IsOsPlatform('Windows'))">
+    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <!-- The MSAL.snk has both private and public keys -->
+    <DelaySign>false</DelaySign>
```
Comment (jmprieur) [inline src/Directory.Build.props:55]: Could go line 26

File: src/Directory.Build.props [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    <AdditionalFiles Include="..\..\.sonarlint\azuread_microsoft-identity-web\CSharp\SonarLint.xml" Link="SonarLint.xml" />
+  </ItemGroup>
+
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Release|AnyCPU'">
+    <DocumentationFile>Microsoft.Identity.Web.xml</DocumentationFile>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TF_BUILD)' == 'true'">
+    <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
+    <PublishRepositoryUrl>true</PublishRepositoryUrl>
```
Comment (jmprieur) [inline src/Directory.Build.props:69]: We already have 4 of these line 32-36

File: src/Directory.Build.props [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    <IncludeSymbols>true</IncludeSymbols>
+    <SymbolPackageFormat>snupkg</SymbolPackageFormat>
+    <EmbedUntrackedSources>true</EmbedUntrackedSources>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net5.0'">
```
Comment (jmprieur) [inline src/Directory.Build.props:75]: Not for this PR, but we could even get rid of these variables as the compiler already defines nice variables
(but we would have to change the source code, of course, to replace the current constants by the compiler defined ones).

See **The complete list of preprocessor symbols for .NET target frameworks** in https://docs.microsoft.com/en-us/dotnet/standard/frameworks

File: src/Directory.Build.props [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    <AzureIdentityVersion>1.3.0</AzureIdentityVersion>
+    <AzureSecurityKeyVaultCertificatesVersion>4.1.0</AzureSecurityKeyVaultCertificatesVersion>
+    <MicrosoftGraphVersion>4.11.0</MicrosoftGraphVersion>
+    <MicrosoftGraphBetaVersion>4.22.0-preview</MicrosoftGraphBetaVersion>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(TargetFramework)' == 'net472' Or '$(TargetFramework)' == 'net462' Or '$(TargetFramework)' == 'net5.0'">
+    <MicrosoftAspNetCoreAuthenticationJwtBearerVersion>5.0.12-*</MicrosoftAspNetCoreAuthenticationJwtBearerVersion>
+    <MicrosoftAspNetCoreAuthenticationOpenIdConnectVersion>5.0.12-*</MicrosoftAspNetCoreAuthenticationOpenIdConnectVersion>
+    <MicrosoftExtensionsCachingMemoryVersion>5.0.0</MicrosoftExtensionsCachingMemoryVersion>
```
Comment (jmprieur) [inline src/Directory.Build.props:109]: For Microsoft.Extension.Caching.Abstractions, I think that we might want to group net472 with netstandard2.0.
@GeoK will confirm

File: src/Directory.Build.props [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+    <MicrosoftExtensionsHostingVersion>5.0.0</MicrosoftExtensionsHostingVersion>
+    <MicrosoftAspNetCoreDataProtectionVersion>5.0.8</MicrosoftAspNetCoreDataProtectionVersion>
+    <MicrosoftExtensionsCachingMemoryVersion>5.0.0</MicrosoftExtensionsCachingMemoryVersion>
```
Comment (jmprieur) [inline src/Directory.Build.props:112]: Duplicate of 109. Maybe one of these is Microsoft.Extensions.Caching.Abstractions?

File: src/Microsoft.Identity.Web/Microsoft.Identity.Web.csproj [hunk 7]
```diff
@@ -122,16 +45,9 @@
...
-    <PackageReference Include="Microsoft.Extensions.Hosting" Version="5.0.0" />
-    <PackageReference Include="Microsoft.AspNetCore.DataProtection" Version="5.0.8" />
-  </ItemGroup>
-
-  <ItemGroup>
-    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.3.0">
-      <PrivateAssets>all</PrivateAssets>
-      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
-    </PackageReference>
+    <PackageReference Include="Microsoft.Extensions.Caching.Memory" Version="$(MicrosoftExtensionsCachingMemoryVersion)" />
```
Comment (jmprieur) [inline src/Microsoft.Identity.Web/Microsoft.Identity.Web.csproj:48]: I'm not sure we need it (as it's in Microsoft.Identity.Web.TokenCache already)?

**Review by jmprieur — APPROVED** [review]
Awesome work, @jennyf19 
It's amazing how small the csproj are now. With just the essential / differences

File: src/Directory.Build.props (hunk 1 above)
Comment (jennyf19) [reply src/Directory.Build.props:25]: good catch. Will remove the one on line 9.

File: src/Directory.Build.props (hunk 4 above)
Comment (jennyf19) [reply src/Directory.Build.props:75]: Will do after.

File: src/Microsoft.Identity.Web/Microsoft.Identity.Web.csproj (hunk 7 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Web/Microsoft.Identity.Web.csproj:48]: Let me try without in a different PR.

**Comment by jennyf19** [comment]
@jmprieur ran a release build w/these changes

**Pull request merged**